ndk/src/signers/nip46/rpc.ts

```diff
@@ -117,7 +121,16 @@ export class NDKNostrRpc extends EventEmitter {
         const remoteUser = this.ndk.getUser({ hexpubkey: remotePubkey });
         const request = { id, method, params };
         const promise = new Promise<NDKRpcResponse>((resolve) => {
-            if (cb) this.once(`response-${id}`, cb);
+            const responseHandler = (response: NDKRpcResponse) => {
```

Comment:
I don't see anything dick-buttish in here but I'm not exactly sure what this `auth_url` flow is so I'm not 100% on this...

Reply:
this is the magical part